trade-finance-manager-api/src/v1/api.js

```diff
@@ -302,7 +302,7 @@ const findFacilitiesByDealId = async (dealId) => {
   }
 };
 
-const updateFacility = async (facilityId, facilityUpdate) => {
+const updateFacility = async ({ facilityId, tfmUpdate, sessionTfmUser, sessionPortalUser, isSystemUpdate }) => {
```

Comment:
I think this feels sligtly odd we're passing in all these parameters when, in fact, some are optional. I wonder if there's a better solution here? 

I've not got the time today to give this much thought, but would something with a global enum be better ie:
```
const updateFacility = async ({ facilityId, tfmUpdate, auditInformation }) => {

...

type AuditInformation
{
    userType: 'tfm' | 'portal' | 'system'
    id: number (there's a nice way we can lock this down so that we cant have a number on system, talk to josh if you like this idea)
}
```

then we could have a `createAuditInformation` function that accepts userType and id and returns a nice audit information object?

This is 5 secs of thought, so happy for hot takes and to be told no because!

Reply:
As discussed on slack will implement
```
type UserInformation = {
  userType: 'tfm' | 'portal';
  id: string
} | {
  userType: 'system'
}
```
Alongside helpers `generateSystemUserInformation`, etc in `tfm-api` and `validateUserInformation` in `dtfs-central-api`